cornell-dti/course-plan
# Post Launch: Vue3 Migration, Slot Names, 20 New Majors and Minors, Bug Fixes

### Summary 

This pull request includes post launch bugs, the addition of 20 new majors/minors, our migration to vue3, the addition of entranceYear in onboarding, slot names for requirements, and FA21 Cornell Course Roster API data.

- [x] Migration to vue3
- [x] Code cleanup for course json generator
- [x] New majors/minors (HD major & minor, LING major & minor, MATH major, Material Science EN major, PAM major, AERO EN minor, Physics major, Physics EN major, STS major, ASTRO major, SPAN major & minor, Food Science major, BS major, ES major, ECE minor, Robotics minor, Health Policy minor)
- [x] Slot names
![Screen Shot 2021-05-04 at 11 21 50 PM](https://user-images.githubusercontent.com/54298311/117093641-83b9eb80-ad2f-11eb-8fe0-0ead10417265.png)

- [x] Resolved post launch identified bugs (total credit denominator, edit semester dropdowns, onboarding required fields bug)
![Screen Shot 2021-05-04 at 11 22 55 PM](https://user-images.githubusercontent.com/54298311/117093678-aa782200-ad2f-11eb-806a-82bef867134a.png)

- [x] Added Catalog Attribute to course data fetched from Cornell Course Roster API
- [x] Renaming requirements bar components to be consistent with requirement computation
- [x] Refactoring & modularization of requirements fulfillment
- [x] entranceYear field added to onboarding process
![Screen Shot 2021-05-04 at 11 23 22 PM](https://user-images.githubusercontent.com/54298311/117093705-b9f76b00-ad2f-11eb-873e-4015f5d858dc.png)

- [x] FA21 course data
![Screen Shot 2021-05-04 at 11 23 51 PM](https://user-images.githubusercontent.com/54298311/117093733-cbd90e00-ad2f-11eb-8e94-ea756b9d7740.png)
- [x] Updated READMe with CP’s new members: Kehui, Miranda, Vaish

Remaining TODOs:

- [ ] Further explore Fuse.js (or another approach) to optimize search while maintaining good performance.

### Test Plan 
1. Try adding courses, adding/editing semesters, using onboarding (note the entrance year addition).
2. You should not be able to move past first page of onboarding without completing required fields.
3. When you add a course that already exists, the total credit denominator should not increase from 120.
4. Should see Slot Names (check Liberal Studies EN requirement).
5. Click a course offered in the Fall (like CS 1110) and see that the info in the Bottom Bar is from FA21.
6. Delete all your semesters, add a semester that's not Spring 2021. Edit the semester. You should not see dupes.
